datavec/datavec-python/src/main/java/org/datavec/python/PythonJob.java

```diff
@@ -0,0 +1,109 @@
+        if (setupRunMode)setup();
+    }
+
+    public void clearState(){
+        PythonContextManager.setContext("main");
+        PythonContextManager.deleteContext(context);
+        setup();
+    }
+
+    public void setup(){
+        try (PythonGIL gil = PythonGIL.lock()){
+            PythonContextManager.setContext(context);
+            PythonObject runF = PythonExecutioner.getVariable("run");
+            if (runF.isNone()){
+                PythonExecutioner.exec(code);
+            }
+            PythonExecutioner.getVariable("run");
+            if (runF.isNone()){
+                throw new RuntimeException("run() method not found!");
+            }
+            this.runF = runF;
+            PythonObject setupF = PythonExecutioner.getVariable("setup");
+            if (!setupF.isNone()){
+                setupF.call();
+            }
+        }
+    }
+    public void exec(PythonVariables inputs, PythonVariables outputs){
+        try (PythonGIL gil = PythonGIL.lock()){
+            PythonContextManager.setContext(context);
+            if (!setupRunMode){
+                PythonExecutioner.exec(code, inputs, outputs);
+                return;
+            }
+            PythonExecutioner.setVariables(inputs);
+            PythonObject inspect = Python.importModule("inspect");
+            PythonObject argsList = inspect.attr("getfullargspec").call(runF).attr("args");
+            PythonObject runargs = Python.dict();
+            int argsCount = Python.len(argsList).toInt();
+            for(int i=0; i <argsCount; i++){
+                PythonObject arg = argsList.get(i);
+                PythonObject val = Python.globals().get(arg);
+                if(val.isNone()){
+                    throw new RuntimeException("Input value not received for run() argument: " + arg.toString());
+                }
+                runargs.set(arg, val);
+            }
+            PythonObject outDict = runF.callWithKargs(runargs);
+            Python.globals().attr("update").call(outDict);
+
+            PythonExecutioner.getVariables(outputs);
+        }
+    }
+
+    public PythonVariables execAndReturnAllVariables(PythonVariables inputs){
+        try (PythonGIL gil = PythonGIL.lock()){
+            if (!setupRunMode){
+                return PythonExecutioner.execAndReturnAllVariables(code, inputs);
+            }
+            PythonExecutioner.setVariables(inputs);
+            PythonObject inspect = Python.importModule("inspect");
+            PythonObject argsList = inspect.attr("getfullargspec").call(runF).attr("args");
+            PythonObject runargs = Python.dict();
+            int argsCount = Python.len(argsList).toInt();
+            for(int i=0; i <argsCount; i++){
+                PythonObject arg = argsList.get(i);
+                PythonObject val = Python.globals().get(arg);
+                if(val.isNone()){
+                    throw new RuntimeException("Input value not received for run() argument: " + arg.toString());
```

Comment:
Create a custom checked exception in python in place of RuntimeException and add a throws clause of that exception please.